src/ansys/dpf/core/fields_container.py

```diff
@@ -590,6 +585,118 @@ def animate(self, save_as=None, deform_by=None, scale_factor=1.0, **kwargs):
             **kwargs,
         )
 
+    def animate_modal(
+        self, mode_number, save_as=None, deform_by=None, frame_number=21, scale_factor=1.0, **kwargs
+    ):
+        """Creates an animation based on the Fields contained in the FieldsContainer.
+
+        This method creates a movie or a gif based on the time ids of a FieldsContainer.
+        For kwargs see pyvista.Plotter.open_movie/add_text/show.
+
+        Parameters
+        ----------
+        save_as : Path of file to save the animation to. Defaults to None. Can be of any format
+            supported by pyvista.Plotter.write_frame (.gif, .mp4, ...).
+        deform_by : FieldsContainer, Result, Operator, optional
+            Used to deform the plotted mesh. Must return a FieldsContainer of the same length as
+            self, containing 3D vector Fields of distances.
+            Defaults to None, which takes self if possible. Set as False to force static animation.
+        scale_factor : float, list, optional
+            Scale factor to apply when warping the mesh. Defaults to 1.0. Can be a list to make
+            scaling frequency-dependent.
+        """
+        from ansys.dpf.core.animator import Animator
+
+        # Create a workflow defining the result to render at each step of the animation
+        wf = dpf.core.Workflow()
+        # First define the workflow index input
+        forward_index = dpf.core.operators.utility.forward()
+        wf.set_input_name("loop_over", forward_index.inputs.any)
+        # Define the field extraction using the fields_container and indices
+        extract_field_op = dpf.core.operators.utility.extract_field(self, indices=[mode_number - 1])
```

Comment:
Here `mode_number` is read as the position (one-based) in the FieldsContainer. I think what should be done is extract the fields for `mode_number` as a value of the TimeFreqSupport. This way, if you print(fields_container) and it shows "time: 45", you can request mode 45. Here mode_number=N means "the Nth field available" and not necessarily the Nth mode.